hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java

```diff
@@ -325,6 +324,40 @@ private void buildApiNodeEdges(StructuredTraceGraph graph) {
           }
         }
       }
+      // Sometimes an exit span might be missing for services like Istio, Kong.
+      // Only Entry spans will be populated for these services,
+      // an edge must be created between these services as well.
+      // 1. get entry boundary event from an api node.
+      // 2. find all the children of entry boundary event, which will be entry boundary nodes of
+      //  different api nodes
+      // 3. Check both parent span and child belong to different service as well.
+      // 4. connect the entry boundary of different api nodes with an edge.
+      Optional<Event> entryBoundaryEvent = apiNode.getEntryApiBoundaryEvent();
+      if (entryBoundaryEvent.isPresent()) {
+        List<Event> children = graph.getChildrenEvents(entryBoundaryEvent.get());
+        if (children != null) {
+          for (Event child : children) {
+            // if the child of an entry boundary event is an entry api boundary type,
+            // which can happen if exit span missing and both belongs to different services.
+            if (EnrichedSpanUtils.isEntryApiBoundary(child)
+                && GraphBuilderUtil.areBothSpansFromDifferentService(
+                    child, entryBoundaryEvent.get())) {
+              ApiNode<Event> destinationApiNode = entryApiBoundaryEventIdToApiNode.get(child);
+              LOGGER.debug(
```

Comment:
Wrap this within `if (LOGGER.isDebugEnabled()) {`

`HexUtils.getHex(` could be expensive and shouldn't be done if debug logs are not enabled.